core/src/BibleEngine.class.ts

```diff
@@ -145,28 +144,22 @@ export class BibleEngine {
         this.pDB = this.createConnection(dbConfig);
     }
 
-    async findConnection(dbConfig: ConnectionOptions) {
-        try {
-            return getConnection(dbConfig.name || this.CONNECTION_NAME).manager;
-        } catch (error: any) {
-            if (error?.name === 'ConnectionNotFoundError') {
-                return this.createConnection(dbConfig);
-            }
-            throw error;
-        }
+    async findConnection(dbConfig: DataSourceOptions) {
+        if (this.dataSource) return this.dataSource.manager;
+        else return this.createConnection(dbConfig);
     }
 
-    async createConnection(dbConfig: ConnectionOptions) {
-        const connection = await createConnection({
+    async createConnection(dbConfig: DataSourceOptions) {
```

Comment:
this appears to be a breaking change, can you do a major version increment?

https://semver.org/#spec-item-8

even though there's some overlapping interface, I'd rather be a bit strict on this and make it clear for consumers of the package!

Reply:
I agree with being strict with breaking changes, have been regularly bitten by libraries breaking code in minor versions.

However: the interface in this case is identical, as far as i see it's just a name change of the type. Taking into account that we are probably currently the only two projects using it in production i wonder if it's better not to go to quickly with the major versions (if the breakage is only of a theoretical nature)?

Reply:
since i just stumbled over it (from the typeorm source):
```typescript
export declare type ConnectionOptions = DataSourceOptions;
```
`ConnectionOptions` is an alias to `DataSourceOptions`, so the interface is fully identical